syntaxes/src/expression/grammar.ts

```diff
@@ -0,0 +1,886 @@
+        },
+        {
+          name: 'keyword.operator.bitwise.ts',
+          match: /\&|~|\^|\|/,
+        },
+        {
+          name: 'keyword.operator.assignment.ts',
+          match: /\=/,
+        },
+        {
+          name: 'keyword.operator.decrement.ts',
+          match: /--/,
+        },
+        {
+          name: 'keyword.operator.increment.ts',
+          match: /\+\+/,
+        },
+        {
+          name: 'keyword.operator.arithmetic.ts',
+          match: /\%|\*|\/|-|\+/,
+        },
+        {
+          match: /(?<=[_$[:alnum:]])\s*(\/)(?![\/*])/,
+          captures: {
+            '1': {
+              name: 'keyword.operator.arithmetic.ts',
+            },
+          },
+        },
+      ],
+    },
+
+    functionCall: {
+      begin: /(?=(\??\.\s*)?([_$[:alpha:]][_$[:alnum:]]*)\s*(<([^<>]|\<[^<>]+\>)+>\s*)?\()/,
+      end: /(?<=\))(?!(\??\.\s*)?([_$[:alpha:]][_$[:alnum:]]*)\s*(<([^<>]|\<[^<>]+\>)+>\s*)?\()/,
+      patterns: [
+        {
+          name: 'punctuation.accessor.ts',
+          match: /\?/,
+        },
+        {
+          name: 'punctuation.accessor.ts',
+          match: /\./,
+        },
+        {
+          name: 'entity.name.function.ts',
+          match: /([_$[:alpha:]][_$[:alnum:]]*)/,
+        },
+        {
+          name: 'meta.type.parameters.ts',
+          begin: /\</,
+          beginCaptures: {
+            '0': {
+              name: 'punctuation.definition.typeparameters.begin.ts',
+            },
+          },
+          end: /\>/,
+          endCaptures: {
+            '0': {
+              name: 'punctuation.definition.typeparameters.end.ts',
+            },
+          },
+          patterns: [
+            {
+              include: '#type',
+            },
+            {
+              include: '#punctuationComma',
+            },
+          ],
+        },
+        {
+          include: '#parenExpression',
+        },
+      ],
+    },
+
+    functionParameters: {
+      name: 'meta.parameters.ts',
+      begin: /\(/,
+      beginCaptures: {
+        '0': {
+          name: 'punctuation.definition.parameters.begin.ts',
+        },
+      },
+      end: /\)/,
+      endCaptures: {
+        '0': {
+          name: 'punctuation.definition.parameters.end.ts',
+        },
+      },
+      patterns: [
+        {
+          include: '#decorator',
+        },
+        {
+          include: '#parameterName',
+        },
+        {
+          include: '#variableInitializer',
+        },
+        {
+          name: 'punctuation.separator.parameter.ts',
+          match: /,/,
+        },
+      ],
+    },
+
+    identifiers: {
+      patterns: [
+        {
+          name: 'support.class.ts',
+          match: /([_$[:alpha:]][_$[:alnum:]]*)(?=\s*\.\s*prototype\b(?!\$))/,
+        },
+        {
+          match: '(?x)(\\?\\.|\\!\\.|\\.)\\s*(?:\n  ([[:upper:]][_$[:digit:][:upper:]]*) |\n  ([_$[:alpha:]][_$[:alnum:]]*)\n)(?=\\s*\\.\\s*[_$[:alpha:]][_$[:alnum:]]*)',
```

Comment:
Is the two spaces intentional?

Reply:
This is residual from the [base file](https://github.com/microsoft/vscode-textmate/blob/8961b1554acbb649bdce0d236ab0fd78180447af/test-cases/themes/syntaxes/TypeScript.tmLanguage.json#L2284). I'll remove it.